### PR TITLE
Add new package for loading tsdoc-config.json file

### PR DESCRIPTION
**Pull request by octogonz**
Initial sketch of the **tsdoc-config.json** feature proposed in [RFC #7](https://github.com/microsoft/tsdoc/issues/7#issuecomment-553681442).

Because the loader relies on NodeJS libraries as well as `resolve` and `ajv`, I put this code in a separate NPM package `@microsoft/tsdoc-config`.  That way `@microsoft/tsdoc` can remain fully self-contained and portable.

@ecraig12345

### CODE REVIEW

File: tsdoc-config/package.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  "name": "@microsoft/tsdoc-config",
+  "version": "0.12.15",
```
Comment (iclanton) [inline tsdoc-config/package.json]: This is a weird version. Is this package expected to stay in lockstep with the `tsdoc` package?

File: tsdoc-config/.npmrc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+registry=https://registry.npmjs.org/
```
Comment (iclanton) [inline tsdoc-config/.npmrc]: Is this file necessary?

File: tsdoc-config/src/TSDocConfigFile.ts [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+      if (baseConfigFile.hasErrors) {
+        this._hasErrors = true;
+      }
+    }
+  }
+
+  private static _findConfigPathForFolder(folderPath: string): string {
+    if (folderPath) {
+      let foundFolder: string = folderPath;
+      for (;;) {
```
Comment (iclanton) [inline tsdoc-config/src/TSDocConfigFile.ts]: Ew.

File: tsdoc-config/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (iclanton) [inline tsdoc-config/tsconfig.json]: You don't want to pull RSC into this repo?

File: tsdoc-config/package.json (hunk 1 above)
Comment (octogonz) [reply tsdoc-config/package.json]: I think that would be best, but I'll set up the version policy in a separate PR, since it may alter the CI scripts.

File: tsdoc-config/.npmrc (hunk 2 above)
Comment (octogonz) [reply tsdoc-config/.npmrc]: This was just copied from the other projects in the repo.  Probably they should all be removed.  But let's deal with that in a separate PR.

File: tsdoc-config/src/TSDocConfigFile.ts (hunk 3 above)
Comment (octogonz) [reply tsdoc-config/src/TSDocConfigFile.ts]: Infinite loops are the One True Loop Style! (1TLS)  do/while are degenerate special cases. 😊

File: tsdoc-config/tsconfig.json (hunk 4 above)
Comment (octogonz) [reply tsdoc-config/tsconfig.json]: It's been useful to have a counterexample of a minimalist toolchain.  I'd consider heft, though.
